flash

```diff
@@ -196,18 +197,28 @@ ${LCYAN}The Scoring System:${NC}\n
 "
 }
 
-while getopts 'hiv' flag; do
+while getopts 'hivp:' flag; do
   case "${flag}" in
     h) print_usage && exit ;;
     i) print_info && exit ;;
     v) echo -e "\n${YELLOW}fla.sh Current Version:${NC} ${RED}1.1${NC}\n" && exit ;;
+    p) [[ $(command -v $OPTARG 2>&1) ]] && PREVIEWER=$OPTARG || { echo "Unable to find previewer $OPTARG. Exiting..." && exit 1 ; } ;;
     *) print_usage && exit 1 ;;
   esac
 done
 
 
-# Show pretty FZF preview of decks using BAT
-    DECK="$(find . -maxdepth "$SEARCH_DEPTH" -iname "*.csv" | fzf --preview='bat --theme="Solarized (dark)" --style=numbers --color=always {} | head -500')"
+# Set some parameters for preview command used by FZF
+while [[ -z $PREVIEWER_PARAMTERS ]]; do
+  case $PREVIEWER in
+    bat) PREVIEWER_PARAMTERS="--theme='Solarized (dark)' --style=numbers --color=always" ;;
+    cat) PREVIEWER_PARAMTERS="--number" ;;
+    *)   echo -e "$PREVIEWER is not a valid previewer. Use 'bat' or 'cat'. Falling back on default..." && sleep 1 && PREVIEWER='bat' ;;
```

Comment:
change `sleep` to `read` so the user can read the message and then hit a key to move forward

Reply:
Can do. If I add read with the `-p` prompt that will add a bashism. Shouldn't matter just letting you know. Also do you want it to be any key or just enter?

Reply:
good thoughts, lets make it the enter key as thats what i use later in the script, and add some text for the user that reads `Press [Enter] to continue...`